analysis/common_variables.py

```diff
@@ -636,6 +638,204 @@ def generate_common_variables(index_date_variable, exposure_end_date_variable, o
         "tmp_out_date_migraine_snomed", "tmp_out_date_migraine_hes", "tmp_out_date_migraine_death",
     ),
 
+#######################################
+# Neurodegenerative History variables # 
+#######################################
+
+## History of any dementia
+    # Dementia
+    # History of Alzheimer's disease
+    # Primary
+    tmp_out_date_history_alzheimer_disease_snomed=patients.with_these_clinical_events(
+        alzheimer_snomed,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+    # HES
+    tmp_out_date_history_alzheimer_disease_hes=patients.admitted_to_hospital(
+        returning="binary_flag",
+        with_these_diagnoses=alzheimer_icd10,
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+
+    # Combined history of alzheimer disease
+    out_date_history_alzheimer_disease = patients.minimum_of(
+        "tmp_out_date_history_alzheimer_disease_snomed", "tmp_out_date_history_alzheimer_disease_hes", 
+    ),
+
+    # History of  vascular dementia
+    # Primary
+    tmp_out_date_history_vascular_dementia_snomed=patients.with_these_clinical_events(
+        vascular_dementia_snomed,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+        },
+    ),
+    # HES
+    tmp_out_date_history_vascular_dementia_hes=patients.admitted_to_hospital(
+        with_these_diagnoses=vascular_dementia_icd10,
+        returning="binary_flag",
+        on_or_before=f"{index_date_variable} - 1 day",
+        return_expectations={"incidence": 0.1,
+         },
+    ),
+
+    # Combined alzheimer disease
```

Comment:
vascular?

Reply:
It is needed to create history of any_dementia.